File: Python/ParkingLot.py
# Saya Akmal Zulkifli, NIM 2101310, mengerjakan soal Latihan 4
# dalam mata kuliah DPBO. Untuk keberkahannya maka saya tidak
# melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

from Car import Car
from Motorcycle import Motorcycle

class ParkingLot:
    def __init__(self, kapasitas):
        self._kapasitas = kapasitas
        self._jumlah_kendaraan_terparkir = 0

    def park_vehicle(self):
        if self._jumlah_kendaraan_terparkir < self._kapasitas:
            self._jumlah_kendaraan_terparkir += 1
            print("Kendaraan berhasil diparkir.")
        else:
            print("Parking lot penuh, tidak dapat memarkir kendaraan.")

    def unpark_vehicle(self):
        if self._jumlah_kendaraan_terparkir > 0:
            self._jumlah_kendaraan_terparkir -= 1
            print("Kendaraan berhasil dikeluarkan.")
        else:
            print("Parking lot kosong, tidak ada kendaraan yang bisa dikeluarkan.")

    def tampilkan_info(self):
        print(f"Parking Lot Kapasitas: {self._kapasitas}")
        print(f"Jumlah Kendaraan Terparkir: {self._jumlah_kendaraan_terparkir}")

File: C++/ParkingLot.cpp
/*
Saya Akmal Zulkifli, NIM 2101310, mengerjakan soal Latihan 4
dalam mata kuliah DPBO. Untuk keberkahannya maka saya tidak
melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
*/
// Pragma once diguanakan agar header file hanya di-include satu kali
#pragma once
// Library yang digunakan
#include <iostream>
#include <string>
#include "Car.cpp"
#include "Motorcycle.cpp"
// Deklarasi namespace
using namespace std;

class ParkingLot
{
private:
    string Kapasitas;
    string JumlahKendaraanTerparkir;
    Car Mobil;
    Motorcycle Motor;

public:
    ParkingLot()
    {
    }

    ParkingLot(string Kapasitas)
    {
        this->Kapasitas = Kapasitas;
        JumlahKendaraanTerparkir = "";
    }

    ParkingLot(string Kapasitas, string JumlahKendaraanTerparkir)
    {
        this->Kapasitas = Kapasitas;
        this->JumlahKendaraanTerparkir = JumlahKendaraanTerparkir;
    }

    ParkingLot(string Kapasitas, string JumlahKendaraanTerparkir, Car Mobil)
    {
        this->Kapasitas = Kapasitas;
        this->JumlahKendaraanTerparkir = JumlahKendaraanTerparkir;
        this->Mobil = Mobil;
    }

    ParkingLot(string Kapasitas, string JumlahKendaraanTerparkir, Car Mobil, Motorcycle Motor)
    {
        this->Kapasitas = Kapasitas;
        this->JumlahKendaraanTerparkir = JumlahKendaraanTerparkir;
        this->Mobil = Mobil;
        this->Motor = Motor;
    }

    void setKapasitas(string Kapasitas)
    {
        this->Kapasitas = Kapasitas;
    }

    string getKapasitas()
    {
        return Kapasitas;
    }

    void setJumlahKendaraanTerparkir(string JumlahKendaraanTerparkir)
    {
        this->JumlahKendaraanTerparkir = JumlahKendaraanTerparkir;
    }

    string getJumlahKendaraanTerparkir()
    {
        return JumlahKendaraanTerparkir;
    }

    void setCar(Car Mobil)
    {
        this->Mobil = Mobil;
    }

    Car getCar()
    {
        return Mobil;
    }

    void setMotorcycle(Motorcycle Motor)
    {
        this->Motor = Motor;
    }

    Motorcycle getMotorcycle()
    {
        return Motor;
    }

    ~ParkingLot()
    {
    }
};
